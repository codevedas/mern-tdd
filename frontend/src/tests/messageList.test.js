import React from 'react'
import MessageList from '../components/MessageList'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow } from 'enzyme'
Enzyme.configure({ adapter: new Adapter()})
describe('List', () => {
it('renders without crashing', () => {
    const component = mount(<MessageList/>);
    expect(component).toMatchSnapshot();
  });
it('takes messages as props and displays them', () => {
  const component = shallow(<MessageList
    messages={[{id:1, content:'hello', date:'2000'},{id:2, content:'world', date:'2001'}]}
    />);
expect(component.find('ul#message_list').children().length).toBe(2);});
});
