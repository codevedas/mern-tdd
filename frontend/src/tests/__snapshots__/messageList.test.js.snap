// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders without crashing 1`] = `ReactWrapper {}`;
